packages/dds/sequence/src/test/attribution/sharedString.attribution.spec.ts

```diff
@@ -0,0 +1,697 @@
+import { SharedString } from "../../sharedString";
+import { SharedStringFactory } from "../../sequenceFactory";
+import { assertConsistent, Client } from "../intervalUtils";
+
+function makeMockAudience(clientIds: string[]): IAudience {
+	const clients = new Map<string, IClient>();
+	clientIds.forEach((clientId, index) => {
+		const stringId = String.fromCharCode(index + 65);
+		const name = stringId.repeat(10);
+		const userId = `${name}@microsoft.com`;
+		const email = userId;
+		const user = {
+			id: userId,
+			name,
+			email,
+		};
+		clients.set(clientId, {
+			mode: "write",
+			details: { capabilities: { interactive: true } },
+			permission: [],
+			user,
+			scopes: [],
+		});
+	});
+	// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+	return {
+		getMember: (clientId: string): IClient | undefined => {
+			return clients.get(clientId);
+		},
+	} as IAudience;
+}
+
+interface FuzzTestState extends BaseFuzzTestState {
+    containerRuntimeFactory: MockContainerRuntimeFactoryForReconnection;
+    clients: Client[];
+	// Note: eventually each client will have an Attributor. While design/implementation is in flux, this test suite
+	// just stores a singleton attributor for the purposes of assessing its size.
+	attributor?: IAttributor;
+	serializer?: Encoder<IAttributor, string>;
+}
+
+interface ClientSpec {
+    stringId: string;
+}
+
+interface RangeSpec {
+    start: number;
+    end: number;
+}
+
+interface AddText extends ClientSpec {
+    type: "addText";
+    index: number;
+    content: string;
+	props?: PropertySet;
+}
+
+interface RemoveRange extends ClientSpec, RangeSpec {
+    type: "removeRange";
+}
+
+interface AnnotateRange extends ClientSpec, RangeSpec {
+    type: "annotateRange";
+    properties: PropertySet;
+}
+
+interface Synchronize {
+    type: "synchronize";
+}
+
+type TextOperation = AddText | RemoveRange | AnnotateRange;
+
+type Operation = TextOperation | Synchronize;
+
+interface OperationGenerationConfig {
+    /**
+     * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+     * Note due to concurency, during test execution the actual length of the string may exceed this.
+     */
+    maxStringLength?: number;
+    /**
+     * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+     * Note due to concurency, during test execution the actual number of intervals may exceed this.
+     */
+    maxIntervals?: number;
+    maxInsertLength?: number;
+    propertyNamePool?: string[];
+    validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+    maxStringLength: 1000,
+    maxIntervals: 100,
+    maxInsertLength: 10,
+    propertyNamePool: ["prop1", "prop2", "prop3"],
+    validateInterval: 100,
+};
+
+function makeOperationGenerator(optionsParam?: OperationGenerationConfig): Generator<Operation, FuzzTestState> {
```

Comment:
a lot of this is duped from interval collection fuzz test infra. Think it's worth factoring out in the PR that introduces it? i'm open to it.

On that note, I could split most of the merge-tree changes into a different PR from the sequence changes (which are basically just test additions).